### PR TITLE
Document hybrid prompt-cache and checkpoint behavior

### PR DESCRIPTION
**Pull request by bri-prism**
## What

Add a prompt-cache troubleshooting guide, linked from the README and agent guide, addressing the configuration and numerical-comparison questions in #147.

## Why

The warning about unsupported `--cache-reuse` refers to chunk shifting, while hybrid models can still reuse context checkpoints with a projector loaded. Explicit launcher flags can also override attempted environment changes. Conflating these controls led to advice that disabled useful prefix reuse.

## How

Explain the separate cache controls, provide single-slot test commands, and distinguish checkpoint restoration from prefill segmentation. Document that `cache_prompt: false` does not disable checkpoint creation, so checkpoint settings can affect fresh-request batch shapes and floating-point results.

## Validation

- Checked the launcher argument forwarding and server checkpoint/batch-splitting paths.
- Local checkpoint on/off, repeat/recompute, and forced-batch controls support the documented distinction. The original reporter's exact output divergence was not reproduced.
- Documentation links, Markdown whitespace, and public-disclosure checks passed.

Documentation only; no launcher defaults or runtime behavior changed. References #147 without claiming to fix a runtime correctness defect.

### CODE REVIEW

File: PROMPT-CACHE.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+`cache_prompt: false` prevents reading a previous prompt's prefix; it does not disable
+the server's checkpoint creation or the associated batch splits. This explains why
+turning checkpoints on can affect a fresh server's first request.
+
+Different batch shapes can change floating-point results. Greedy sampling does not
+guarantee identical output across those shapes: a small logit change can select a
+different token and alter the continuation. The reporter's matching long-prompt
+outputs are useful evidence for those requests, not a guarantee for every length.
+
+For short independent requests, compare this explicitly with checkpoints enabled:
```
[the remark this answers is not in the document]
Comment (bri-prism) [reply PROMPT-CACHE.md]: Fixed in 86932e0. You were right that the sentence and the command disagreed. The command passes `--ctx-checkpoints 0`, so it is the disabled arm, and the line now says so. That also makes it consistent with the paragraph below telling you to vary only this flag.

**Review by khosravipasha — COMMENTED** [review]
**Agent review** (checked against the server source at the current `prism` head, 6ac5eb087, and the public launcher `scripts/start_llama_server.sh`).

The technical content holds up. Verified: `--cache-reuse` and context checkpoints are separate mechanisms and the reuse warning fires for multimodal and again for non-shiftable memory; explicit CLI flags override `LLAMA_ARG_*`; `cache_prompt: false` only gates reading a cached prefix and does not affect checkpoint creation (creation depends on `--ctx-checkpoints`, task type and memory type); the end-of-prompt split is real, the server breaks the last batch 4 tokens and 4 + n_ubatch tokens before the end so a checkpoint can be taken, plus at the start of the last user message. The advice to compare logits rather than output hashes, and not to adopt `--cache-ram 0 --ctx-checkpoints 0` as a workaround, is right.

Suggestions before merging:

1. **State the defaults and that the demo already has them.** The server ships with `--cache-ram 8192` (MiB), `--ctx-checkpoints 32` (`--checkpoint-min-step 8192`), `--cache-idle-slots` enabled, `--cache-reuse 0`, request `cache_prompt: true`, `--parallel 1`. Our launcher passes none of these, so checkpoints and an 8 GiB prompt cache are on out of the box. As written, the test command reads like a recipe users must apply, while it only lowers the budget to 4 GiB and adds two flags that are already defaults. A Default column in the controls table would carry most of this.
2. **The 8 GiB budget is fixed, not adaptive.** It is the same on a 16 GB laptop and a 192 GB workstation. The cache is not preallocated and evicts oldest entries at the limit, but the only runtime adaptation is on `std::bad_alloc` (limit cut to 40% of the current size), which macOS never reaches because it swaps first. On a 16 GB machine running the 27B (6.7 GiB weights plus live KV) the cache can push into swap without any warning. Worth a paragraph with per-RAM guidance: about a quarter of physical RAM for `--cache-ram`, and 0 on 16 GB. Each cached 27B entry costs about 64 KiB per token at FP16 KV (16 KiB at 4-bit), so 8 GiB holds roughly 128K tokens of history at FP16.
3. "In the revision reported in #147, one boundary is four tokens before the end" implies it changed since; it is the same at head. Drop the revision qualifier and mention the second boundary at 4 + n_ubatch (516 tokens at the default ubatch).
4. Say plainly that `--cache-reuse` is always off for Bonsai: the hybrid GDN memory cannot shift, with or without a projector, so nobody tries to enable it.
5. README placement: the link is the first line of Quick Start, ahead of the two setup commands. It fits better under troubleshooting or next to the AGENTS.md note.
6. `--parallel 1` in both example commands is the default and can go.

A RAM-relative `--cache-ram` default in the launcher would address point 2 properly; that is a runtime change and belongs in a separate PR.
